sdk/lib/opentelemetry/sdk/internal.rb

```diff
@@ -44,13 +44,13 @@ def valid_value?(value)
         valid_simple_value?(value) || valid_array_value?(value)
       end
 
-      def valid_attributes?(attrs)
+      def valid_attributes?(owner, kind, attrs)
         attrs.nil? || attrs.all? do |k, v|
           if !valid_key?(k)
-            OpenTelemetry.handle_error(message: "invalid attribute key type #{k.class}")
+            OpenTelemetry.handle_error(message: "invalid #{kind} attribute key type #{k.class} on span '#{owner}'")
             false
           elsif !valid_value?(v)
-            OpenTelemetry.handle_error(message: "invalid attribute value type #{v.class}")
+            OpenTelemetry.handle_error(message: "invalid #{kind} attribute value type #{v.class} on span '#{owner}'")
```

Comment:
I think it would be valuable to identify which key has a nil value associated with it.

Reply:
🤦 of course

Reply:
done